File: archives/C#/0290. Word Pattern.cs
public class Solution {
    public bool WordPattern(string pattern, string str) {
        string[] stringStr=str.Split(" ");
        if(pattern.Length!=stringStr.Length){
            return false;
        }
        Dictionary<char,string> dictCmp=new Dictionary<char,string>();
        HashSet<string> strSet=new HashSet<string>();
        bool flag=true;
        for(int i=0;i<pattern.Length;i++){
            if(!dictCmp.ContainsKey(pattern[i])){
                if (!strSet.Contains(stringStr[i])){
                    dictCmp[pattern[i]]=stringStr[i];
                    strSet.Add(stringStr[i]);
                }
                else{
                    flag=false;
                    break;
                }
            }
            else{
                if(dictCmp[pattern[i]]!=stringStr[i]){
                    flag=false;
                    break;
                }
            }
        }
        return flag;
    }
}

File: archives/leetcode2/0290. Word Pattern.py
class Solution(object):
    def wordPattern(self, pattern, str):
        """
        :type pattern: str
        :type str: str
        :rtype: bool
        """
        strlist=str.split(' ')
        if len(pattern)!=len(strlist):
            return False
        dictpattern={}
        strlistset=set()
        for k in range(len(pattern)):
            if pattern[k] not in dictpattern:
                if strlist[k] in strlistset:
                    return False
                else:
                    dictpattern[pattern[k]]=strlist[k]
                    strlistset.add(strlist[k])
            else:
                if dictpattern[pattern[k]]!=strlist[k]:
                    return False
        return True
